### PR TITLE
Tone down walreceiver connection timeout errors

### PR DESCRIPTION
**Pull request by SomeoneToIgnore**
Closes https://github.com/neondatabase/neon/issues/3342

### CODE REVIEW

File: pageserver/src/walreceiver/walreceiver_connection.rs [hunk 1]
```diff
@@ -77,9 +77,13 @@ pub async fn handle_walreceiver_connection(
...
                 return Ok(());
             }
-            Err(elapsed) => anyhow::bail!(
-                "Timed out while waiting {elapsed} for walreceiver connection to open"
-            ),
+            Err(elapsed) => {
+                // Timing out to connect to a safekeeper node could happen long time, due to
+                // many reasons that pageserver cannot control.
+                // Do not produce an error, but make it visible, that timeouts happen by logging the `event.
+                info!("Timed out while waiting {elapsed} for walreceiver connection to open");
```
Comment (koivunej) [inline pageserver/src/walreceiver/walreceiver_connection.rs:84]: Had to test, what does display on elapsed return: `deadline has elapsed`. Is this intended vs. using `{connect_timeout}` here?

**Review by koivunej — APPROVED** [review]
Just small logging related nit, otherwise looking good.

File: pageserver/src/walreceiver/walreceiver_connection.rs (hunk 1 above)
Comment (funbringer) [reply pageserver/src/walreceiver/walreceiver_connection.rs:84]: Maybe there should be a metric for this? It's nice to have metrics for bad situations which might spiral out of control.

Comment (SomeoneToIgnore) [reply pageserver/src/walreceiver/walreceiver_connection.rs:84]: Thank you for spotting, timeout value is indeed what's supposed to be printed, changed.

I agree that here it should be a metric, so we can always see whether we're connected to something and when the last such event had happened.

I planned to use metrics as part of https://github.com/neondatabase/neon/issues/2106 and keep these PRs small for now.

**Pull request merged**